src/storage/meta.rs

```diff
@@ -0,0 +1,39 @@
+
+use std::ops::{Deref, DerefMut};
+
+use hibitset::{BitSet, BitSetAnd, BitSetLike};
+
+use storage::{MaskedStorage, Storage, WrappedStorage, UnprotectedStorage};
+
+use Index;
+
+/// Observes interactions with the component's storage and stores state alongside it.
+///
+/// Useful for things like tracking modifications to components, storing sorted lists related
+/// to the storage, etc.
+pub trait Metadata<T>: Default {
```

Comment:
When you have a `TrackedStorage`, doesn't that allow to do basically 99% of the meta stuff you need?

Reply:
Sorted lists could be useful as well, as well as wanting to use something other than `TrackedStorage`.

Reply:
Sorted lists can be built using `TrackedStorage`, that's one of my main motivations for adding it.